### PR TITLE
Separate remote and local commands

### PR DESCRIPTION
**Pull request by baude**
In the previous CLI, we had an accurate depiction of commands
available for the remote client and those available for the
local client.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podman/image.go [hunk 1]
```diff
@@ -20,16 +20,16 @@ var (
 var imageSubCommands = []*cobra.Command{
 	_historyCommand,
 	_imageExistsCommand,
-	_inspectCommand,
 	_imagesCommand,
+	_importCommand,
+	_inspectCommand,
 	_pruneImagesCommand,
-	_pushCommand,
```
Comment (rhatdan) [inline cmd/podman/image.go]: Did you mean to remove pushCommand?

Comment (baude) [reply cmd/podman/image.go]: yes, push is not avail for remote I think

**Comment by baude** [comment]
can someone pull the trigger on this?

File: cmd/podman/container.go [hunk 2]
```diff
@@ -15,7 +15,12 @@ var containerCommand = cliconfig.PodmanCommand{
 	},
 }
 
+var containerCommands = []*cobra.Command{
```
Comment (mheon) [inline cmd/podman/container.go]: Why is this separate from the rest? The single exists command, and then `getContainerSubCommands()` separately?

File: cmd/podman/commands.go [hunk 3]
```diff
@@ -6,13 +6,53 @@ import (
 	"github.com/spf13/cobra"
 )
 
+func getRootCommands() []*cobra.Command {
```
Comment (mheon) [inline cmd/podman/commands.go]: Can we rename this? This makes me think of commands that won't run as rootless

File: cmd/podman/diff.go [hunk 4]
```diff
@@ -59,9 +59,8 @@ func init() {
 
 	flags.MarkHidden("archive")
```
Comment (mheon) [inline cmd/podman/diff.go]: Extra newline

File: cmd/podman/images.go [hunk 5]
```diff
@@ -114,8 +114,6 @@ func init() {
 	flags.BoolVarP(&imagesCommand.Quiet, "quiet", "q", false, "Display only image IDs")
 	flags.StringVar(&imagesCommand.Sort, "sort", "created", "Sort by created, id, repository, size, or tag")
```
Comment (mheon) [inline cmd/podman/images.go]: Extra newline

File: cmd/podman/import.go [hunk 6]
```diff
@@ -36,7 +36,6 @@ func init() {
 	flags.StringVarP(&importCommand.Message, "message", "m", "", "Set commit message for imported image")
 	flags.BoolVarP(&importCommand.Quiet, "quiet", "q", false, "Suppress output")
```
Comment (mheon) [inline cmd/podman/import.go]: Extra newline

File: cmd/podman/info.go [hunk 7]
```diff
@@ -37,7 +37,6 @@ func init() {
 	flags.BoolVarP(&infoCommand.Debug, "debug", "D", false, "Display additional debug information")
 	flags.StringVarP(&infoCommand.Format, "format", "f", "", "Change the output format to JSON or a Go template")
```
Comment (mheon) [inline cmd/podman/info.go]: Extra newline

File: cmd/podman/inspect.go [hunk 8]
```diff
@@ -46,7 +46,6 @@ func init() {
 	flags.BoolVarP(&inspectCommand.Latest, "latest", "l", false, "Act on the latest container podman is aware of if the type is a container")
 	flags.BoolVarP(&inspectCommand.Size, "size", "s", false, "Display total file size if the type is container")
```
Comment (mheon) [inline cmd/podman/inspect.go]: Newline

File: cmd/podman/kill.go [hunk 9]
```diff
@@ -40,7 +40,6 @@ func init() {
 	flags.StringVarP(&killCommand.Signal, "signal", "s", "KILL", "Signal to send to the container")
 	flags.BoolVarP(&killCommand.Latest, "latest", "l", false, "Act on the latest container podman is aware of")
```
Comment (mheon) [inline cmd/podman/kill.go]: Newline

File: cmd/podman/load.go [hunk 10]
```diff
@@ -39,7 +39,6 @@ func init() {
 	flags.BoolVarP(&loadCommand.Quiet, "quiet", "q", false, "Suppress the output")
 	flags.StringVar(&loadCommand.SignaturePolicy, "signature-policy", "", "Pathname of signature policy file (not usually used)")
```
Comment (mheon) [inline cmd/podman/load.go]: Newline

File: cmd/podman/login.go [hunk 11]
```diff
@@ -44,7 +44,6 @@ func init() {
 	flags.BoolVar(&loginCommand.TlsVerify, "tls-verify", true, "Require HTTPS and verify certificates when contacting registries (default: true)")
 	flags.StringVarP(&loginCommand.Username, "username", "u", "", "Username for registry")
```
Comment (mheon) [inline cmd/podman/login.go]: Newline

File: cmd/podman/logout.go [hunk 12]
```diff
@@ -32,7 +32,6 @@ func init() {
 	flags.BoolVarP(&logoutCommand.All, "all", "a", false, "Remove the cached credentials for all registries in the auth file")
 	flags.StringVar(&logoutCommand.Authfile, "authfile", "", "Path of the authentication file. Default is ${XDG_RUNTIME_DIR}/containers/auth.json. Use REGISTRY_AUTH_FILE environment variable to override")
```
Comment (mheon) [inline cmd/podman/logout.go]: Newline

File: cmd/podman/logs.go [hunk 13]
```diff
@@ -43,7 +43,6 @@ func init() {
 
 	flags.SetInterspersed(false)
```
Comment (mheon) [inline cmd/podman/logs.go]: Newline

File: cmd/podman/main.go [hunk 14]
```diff
@@ -92,6 +108,8 @@ func init() {
 	rootCmd.PersistentFlags().BoolVar(&MainGlobalOpts.Syslog, "syslog", false, "Output logging information to syslog as well as the console")
 
 	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.TmpDir, "tmpdir", "", "Path to the tmp directory")
+	rootCmd.AddCommand(rootCommands...)
```
Comment (mheon) [inline cmd/podman/main.go]: So how are these different?

Comment (mheon) [reply cmd/podman/main.go]: If one of these is remote and one is not, can we document that somehow? Because I have no idea how to tell which is which

**Comment by mheon** [comment]
I don't really care about all the newlines, fix them if you have time, but this really needs a few renames and some comments explaining what's going on

File: cmd/podman/container.go (hunk 2 above)
Comment (baude) [reply cmd/podman/container.go]: added a comment but for the record, this is the way the old cli was too. commands that both the remote client and local client have implemented are there.

**Comment by baude** [comment]
@mheon clarified here and there, renamed where you suggested, the newlines i get in a diff PR that is waiting on this one.

**Comment by mheon** [comment]
Alright, LGTM

**Comment by mheon** [comment]
@rhatdan @TomSweeneyRedHat @vrothberg PTAL

**Comment by baude** [comment]
bot, retest this please

**Comment by TomSweeneyRedHat** [comment]
LGTM, but will you have a merge dance with #2309?

**Comment by baude** [comment]
depends, merge mine and we will find out

**Pull request merged**